pandas/tests/frame/test_alter_axes.py

```diff
@@ -126,21 +131,29 @@ def test_set_index_pass_single_array(self, frame_of_index_cols,
         df.index.name = index_name
 
         key = box(df['B'])
-        # np.array and list "forget" the name of B
-        name = [None if box in [np.array, list] else 'B']
+        if box == list:
```

Comment:
why exactly is a list interpreted differently here? this makes this test really really odd. I am worried something changed here, as this appears to work just fine in the master.

Reply:
You will see above (line 123)  that `list` wasn't tested - for precisely this reason. `df.set_index(['A', 'B'])` interprets a `A` & `B` as column keys, so (assuming this `df` was length 2) it would not use the list `['A', 'B']` as the index. To do that, one would have to pass `[['A', 'B']]`. This PR proposes to add tests for the current behaviour.

Reply:
@jreback 
In case the above was not very clearly worded, this corresponds exactly to behaviour on `master`:
```
>>> df = pd.DataFrame(np.random.randn(2,3))
>>> df.set_index(['A', 'B'])
KeyError: 'A'
>>> df.set_index([['A', 'B']])
          0         1         2
A  1.962370 -1.150770  0.843600
B -0.417274  0.509781 -0.697802
```